### ISSUE TITLE
Our self-trained model has a high WER on English

### ISSUE THREAD

**Issue by wzh172352716** [roudimit/whisper-flamingo#10]
Hi,
Based on the audio_multi_small.yaml configuration, I attempted to fine-tune the Whisper model. However, the test results for the English language showed an error rate exceeding 100+, regardless of whether the audio was clean or noisy. In contrast, the test results for other languages in a clean environment were generally consistent with those reported in the paper. Do you have any experience or insights that could help me?

**Comment by roudimit** [comment]
Hi, thanks for sharing your results. That is strange. First, can you try decoding our pre-trained model to check if the English data is setup well? Also, can you show the training curves from Tensorboard? They should all show smooth progress during training.

**Comment by wzh172352716** [comment]
Thanks for your reply! I checked again and found that the issue was with my English dataset.  

However, when I tried using the model (`whisper_multi-all_small.pt`) you provided to run the English test data (also provided by you), I obtained a WER of 3.9232. In contrast, the result you posted was WER = 1.1.

**Comment by wzh172352716** [comment]
For sure :>
# Select model checkpoint
# Audio checkpoints
# checkpoint=whisper_multi-all_medium.pt  
checkpoint=whisper_multi-all_small.pt  
# checkpoint=checkpoint/whisper_multi-all_small/last-v3.ckpt
# Audio-visual checkpoints
# checkpoint=whisper-flamingo_multi-all_medium.pt
# checkpoint=whisper-flamingo_multi-all_small.pt

# Select model size
# model=large-v2
# model=medium
model=small

# Select modalities
modalities=asr # whisper
# modalities=avsr # whisper-flamingo

# Select av_fusion type
av_fusion="None" # asr - use this for audio only whisper models
# av_fusion="separate" # use this for whisper-flamingo models

# Select whether to use AV-HuBERT encoder
use_av_hubert_encoder=0 # for whisper / asr
# use_av_hubert_encoder=1 # for whisper-flamingo / avsr

# Select multilingual or EN babble noise
# noise_fn=noise/babble/muavic/test.tsv # multilingual babble noise 
noise_fn=/beegfs/data/shared/lrs3/noise/babble/test.tsv # single lrs3 mixture
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/babble/test.tsv
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/music/test.tsv
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/noise/test.tsv 
# noise_fn=/data/sls/scratch/roudi/datasets/lrs3/noise/speech/test.tsv

# Select AV-HuBERT checkpoint
av_hubert_ckpt=models/mavhubert_only_weights.pt # multilingual
# av_hubert_ckpt=models/large_noise_pt_noise_ft_433h_only_weights.pt # english

# Specify Paths
checkpoint_root=models/
# checkpoint_root=models/checkpoint/
checkpoint_path=${checkpoint_root}${checkpoint}
decode_path=decode/
fp16=1
av_hubert_path=/home/wu/work/av_hubert/avhubert

# for ASR only: ignore checkpoint path and use original whisper weights
use_original_whisper=0
# use_original_whisper=1

task=transcribe # ASR
normalizer=fairseq

# for lang in en ar de el es fr it pt ru; do # all langs
# for lang in en es fr it pt; do # high resource langs
# for lang in ar de el ru; do # low resource langs
for lang in en; do
# for beam_size in 1; do
for beam_size in 5; do
    for noise_snr in 0 1000; do
    # for noise_snr in 1000; do
    # for noise_snr in -10 -5 0 5 10; do
            echo $modalities $lang $noise_snr
            sbatch slurm/whisper_decode.sh $lang \
                                    $model \
                                    $noise_snr \
                                    $noise_fn \
                                    $checkpoint_path \
                                    $beam_size \
                                    $modalities \
                                    $use_av_hubert_encoder \
                                    $av_fusion \
                                    $fp16 \
                                    $decode_path \
                                    $av_hubert_path \
                                    $av_hubert_ckpt \
                                    $task \
                                    $normalizer \
                                    $use_original_whisper
        done
    done
done

**Comment by wzh172352716** [comment]
Here is the whisper_decode_multi_wrapper.sh
`# Select model checkpoint
# Audio checkpoints
# checkpoint=whisper_multi-all_medium.pt  
checkpoint=whisper_multi-all_small.pt  
# checkpoint=checkpoint/whisper_multi-all_small/last-v3.ckpt
# Audio-visual checkpoints
# checkpoint=whisper-flamingo_multi-all_medium.pt
# checkpoint=whisper-flamingo_multi-all_small.pt

# Select model size
# model=large-v2
# model=medium
model=small

# Select modalities
modalities=asr # whisper
# modalities=avsr # whisper-flamingo

# Select av_fusion type
av_fusion="None" # asr - use this for audio only whisper models
# av_fusion="separate" # use this for whisper-flamingo models

# Select whether to use AV-HuBERT encoder
use_av_hubert_encoder=0 # for whisper / asr
# use_av_hubert_encoder=1 # for whisper-flamingo / avsr

# Select multilingual or EN babble noise
# noise_fn=noise/babble/muavic/test.tsv # multilingual babble noise 
noise_fn=/beegfs/data/shared/lrs3/noise/babble/test.tsv # single lrs3 mixture
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/babble/test.tsv
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/music/test.tsv
# noise_fn=/data/sls/scratch/roudi/datasets/musan/tsv/noise/test.tsv 
# noise_fn=/data/sls/scratch/roudi/datasets/lrs3/noise/speech/test.tsv

# Select AV-HuBERT checkpoint
av_hubert_ckpt=models/mavhubert_only_weights.pt # multilingual
# av_hubert_ckpt=models/large_noise_pt_noise_ft_433h_only_weights.pt # english

# Specify Paths
checkpoint_root=models/
# checkpoint_root=models/checkpoint/
checkpoint_path=${checkpoint_root}${checkpoint}
decode_path=decode/
fp16=1
av_hubert_path=/home/wu/work/av_hubert/avhubert

# for ASR only: ignore checkpoint path and use original whisper weights
use_original_whisper=0
# use_original_whisper=1

task=transcribe # ASR
normalizer=fairseq

# for lang in en ar de el es fr it pt ru; do # all langs
# for lang in en es fr it pt; do # high resource langs
# for lang in ar de el ru; do # low resource langs
for lang in en; do
# for beam_size in 1; do
for beam_size in 5; do
    for noise_snr in 0 1000; do
    # for noise_snr in 1000; do
    # for noise_snr in -10 -5 0 5 10; do
            echo $modalities $lang $noise_snr
            sbatch slurm/whisper_decode.sh $lang \
                                    $model \
                                    $noise_snr \
                                    $noise_fn \
                                    $checkpoint_path \
                                    $beam_size \
                                    $modalities \
                                    $use_av_hubert_encoder \
                                    $av_fusion \
                                    $fp16 \
                                    $decode_path \
                                    $av_hubert_path \
                                    $av_hubert_ckpt \
                                    $task \
                                    $normalizer \
                                    $use_original_whisper
        done
    done
done`

**Comment by wzh172352716** [comment]
Here is the whisper_decode_multi_wrapper.sh

checkpoint=whisper_multi-all_small.pt  
model=small
modalities=asr # whisper
av_fusion="None" # asr - use this for audio only whisper models
use_av_hubert_encoder=0 # for whisper / asr
noise_fn=/beegfs/data/shared/lrs3/noise/babble/test.tsv # single lrs3 mixture
av_hubert_ckpt=models/mavhubert_only_weights.pt # multilingual
checkpoint_root=models/
checkpoint_path=${checkpoint_root}${checkpoint}
decode_path=decode/
fp16=1
av_hubert_path=/home/wu/work/av_hubert/avhubert
use_original_whisper=0
task=transcribe # ASR
normalizer=fairseq
for lang in en; do
for beam_size in 5; do
    for noise_snr in 0 1000; do

**Comment by roudimit** [comment]
I forgot to note that for the English decoding in clean setting we use `beam_size=1` since the performance with beam search is much worse. For noisy, we still use `beam_size=5`. Also, for all other languages, we use `beam_size=5` for clean and noisy.

Here are the English numbers for `whisper_multi-all_small.pt`:
Clean:
- Beam 1: 1.1
- Beam 5: 3.9

Noisy:
- Beam 1: 17.5
- Beam 5: 16.0

**Comment by roudimit** [comment]
There are a couple things to check. First, if you download our pre-trained audio-visual model (`whisper-flamingo_multi-all_small.pt`) and run decoding, what's the WER with beam size 1 / 5? 
If that works, I think the problem could be your training data. I would start by training the audio model `whisper_multi-all_small` from scratch and ensuring the WER is the same as our pre-trained model.

**Comment by wzh172352716** [comment]
Hi roudimit,

your pre-trained audio-model with the beam-size you mentioned is expected:

clean:1.1
noise:16.9

But when we try to fine-tune the mwhisper based on the babble noise we could get the result 

clean:1.9
noise:17.4

if based on the all noise we could only get the result:

noise:61.32

so you were based on the babble noise to fine-tune the whisper?

**Comment by roudimit** [comment]
We followed the [instructions in AV-HuBERT](https://github.com/facebookresearch/av_hubert/tree/main/avhubert/preparation#audio-noise-preparation-optional) to prepare MUSAN and LRS3 Noise
For training, we use ${musan}/tsv/all including MUSAN babble, music, noise and LRS3 speech.
Does that answer your question?
